src/python/gem5/simulate/simulator.py

```diff
@@ -185,6 +188,42 @@ def unique_exit_event():
 
         self._checkpoint_path = checkpoint_path
 
+    def schedule_simpoint(
+        self,
+        simpoint_start_insts: List[int] = [],
+        schedule_at_init: bool = False
+        )-> None:
+        """
+        Schedule SIMPOINT_BEGIN exit events
+
+        **Warning:** SimPoints only work with one core
+
+        :param simpoint_start_insts: a list of number of instructions
+        indicating the starting point of the simpoints
+        :param schedule_at_init: if it is True, schedule the events in the init
+        stage of the core, else, schedule the events during the simulation
+        """
+        if self._board.get_processor().get_num_cores() > 1:
+            warn("SimPoints only work with one core")
```

Comment:
Should this just be a warning? Maybe it should be a proper exception? A warning won't stop a user doing this, but maybe we should.

Reply:
Good question. It won't *break* anything to have more than 1 core. I think a warning is the right level.